src/LibVLCSharp.Forms/Platforms/Android/VideoViewRenderer.cs

```diff
@@ -23,6 +25,19 @@ public VideoViewRenderer(Context context) : base(context)
         {
         }
 
+        /// <summary>
+        /// Gets triggered when the object gets disposed
+        /// </summary>
+        /// <param name="disposing"></param>
+        protected override void Dispose(bool disposing)
+        {
+            _isDisposing = true;
+            base.Dispose(disposing);
+
+            Control!.MediaPlayer = null;
+            Control!.Dispose();
```

Comment:
It is very _rare_ for ViewRenderers to dispose of the `Control`. Not sure if this is a common leak or if it should not be done.

Reply:
Basically it's enough to set the "_isDisposing here. Just avoid to update the MediaPlayer in the OnMediaPlayerChanging event. I just thought to do also the cleanup here for the Control.

Reply:
Can you check whether Xamarin.Forms performs the `Dispose` internally on the native `Control`? It's not obvious from some glances, but I think it might not. In which case we have a leak, including in the Apple renderer. If so, feel free to update it as well.

Reply:
I try to figure out where the `Control` gets disposed and report back. I never had such issues on iOS, only Android.
Also, all is working fine if you do not use `Shell`. So I guess it's the way how `Xamarin.Forms` handles the page re-creation if you are using `Shell`.

Reply:
I guess this answers the disposing behaviour of `Shell`. @PureWeen wrote this on a `Xamarin.Forms` bug.

> It should be a lot better right now in MAUI. One of the issues here is that on XF shell aggressively disposes and recreates renderers every time you move between tabs/flyout items. With MAUI we aren't doing that and as you move around the native stacks are all maintained.
> 
> If anyone is up for testing this out a bit on MAUI let me know your results.

https://github.com/xamarin/Xamarin.Forms/issues/7521#issuecomment-903285770

So I guess, when the Page is disposing, the MediaPlayerElement on it gets disposed as well (so the native control).
Then, `Shell` recreates the page.

Reply:
The VideoView gets disposed properly already, from my own testing.

Reply:
Ok, then it should be fine to just avoid updating the MediaPlayer element if the renderer is disposing? (Set the flag _isDisposing = true)?

Reply:
There is no need to avoid something if it does not happen in the first place. This really shouldn't happen in the first place, but it's too easy to place the blame on Xamarin Shell only. 

Do you think it could be possible to reproduce without Shell?